apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/util/UserPasswordChecker.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.portal.util;
+
+import com.google.common.base.Strings;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class UserPasswordChecker {
+
+  private static final Pattern PWD_PATTERN = Pattern
```

Comment:
Would be nice if you could write few test cases that check if this pattern really works if intended. Regex is often a reason for bugs.

Reply:
+1

Reply:
Ok, done.